BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -32,48 +33,384 @@ public class SwapTokenStore: ObservableObject {
+        return
+      }
+      guard let response = swapResponse else { return }
+      let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: 16))
+      let gasPrice = "0x\(weiFormatter.weiString(from: response.gasPrice, radix: .hex, decimals: 16) ?? "0")"
+      let gasLimit = "0x\(weiFormatter.weiString(from: response.estimatedGas, radix: .decimal, decimals: 16) ?? "0")"
+      let value = "0x\(weiFormatter.weiString(from: response.value, radix: .decimal, decimals: 16) ?? "0")"
+      let txData: BraveWallet.TxData = .init(
+        nonce: "",
+        gasPrice: gasPrice,
+        gasLimit: gasLimit,
+        to: response.to,
+        value: value,
+        data: self.stringToBytes(response.data))
+      self.transactionController.addUnapprovedTransaction(txData, from: accountInfo.address) { success, txMetaId, error in
+        // should be observed
+      }
+    }
+  }
+  
+  func fetchPriceQuote(base: SwapParamsBase) {
+    guard let swapParams = generateSwapParams(base) else { return }
+    
+    updatingPriceQuote = true
+    swapController.priceQuote(swapParams) { [weak self] success, swapResponse, error in
+      guard success else {
+        self?.clearAllAmount()
+        self?.updatingPriceQuote = false
+        return
+      }
+      self?.updateSwapControls(swapResponse: swapResponse, base: base)
+      self?.updatingPriceQuote = false
+    }
   }
   
-  func prepare(with accountInfo: BraveWallet.AccountInfo) {
+  private func generateSwapParams(_ base: SwapParamsBase) -> BraveWallet.SwapParams? {
+    guard
+      let accountInfo = accountInfo,
+      let sellToken = selectedFromToken,
+      let buyToken = selectedToToken
+    else { return nil }
+    
+    let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: 18))
+    let sellAddress = sellToken.isETH ? BraveWallet.ethSwapAddress : sellToken.contractAddress
+    let buyAddress = buyToken.isETH ? BraveWallet.ethSwapAddress : buyToken.contractAddress
+    let sellAmountInWei: String
+    let buyAmountInWei: String
+    switch base {
+    case .perSellAsset:
+      sellAmountInWei = weiFormatter.weiString(from: sellAmount, radix: .decimal, decimals: Int(sellToken.decimals)) ?? "0"
+      buyAmountInWei = ""
+    case .perBuyAsset:
+      sellAmountInWei = ""
+      buyAmountInWei = weiFormatter.weiString(from: buyAmount, radix: .decimal, decimals: Int(buyToken.decimals)) ?? "0"
+    }
+    let swapParams = BraveWallet.SwapParams(
+      takerAddress: accountInfo.address,
+      sellAmount: sellAmountInWei,
+      buyAmount: buyAmountInWei,
+      buyToken: buyAddress,
+      sellToken: sellAddress,
+      slippagePercentage: slippage,
+      gasPrice: ""
+    )
+    
+    return swapParams
+  }
+  
+  private func clearAllAmount() {
+    sellAmount = "0"
+    buyAmount = "0"
+    selectedFromTokenPrice = "0"
+  }
+  
+  private func updateSwapControls(swapResponse: BraveWallet.SwapResponse?, base: SwapParamsBase) {
```

Comment:
Nit: Don't have arguments that are required for the function to do something optional, do the check at the call site instead. 
Nit: What "Swap controls" are we updating here? Needs some better naming or doc, i.e. `handleSwapResponse(_:base:)`